pkg/logql/log/labels.go

```diff
@@ -495,22 +495,21 @@ func (b *LabelsBuilder) IntoMap(m map[string]string) {
 	}
 }
 
-func (b *LabelsBuilder) Map() map[string]string {
+func (b *LabelsBuilder) Map() (map[string]string, bool) {
 	if !b.hasDel() && !b.hasAdd() && !b.HasErr() {
 		if b.baseMap == nil {
 			b.baseMap = b.base.Map()
 		}
-		return b.baseMap
+		return b.baseMap, false
 	}
 	b.buf = b.UnsortedLabels(b.buf)
 	// todo should we also cache maps since limited by the result ?
 	// Maps also don't create a copy of the labels.
```

Comment:
Is this still relevant?

Reply:
I think so, yes. It was added in Cyrils PR that added the `baseMap` in the first place, and I haven't changed that behaviour at all: https://github.com/grafana/loki/pull/6110/commits/e6002637698e89b1fe888005dd0f46fcd00ba676